lib/active_record_shards/connection_switcher.rb

```diff
@@ -126,7 +129,7 @@ def on_cx_switch_block(which, force: false, construct_ro_scope: nil, &block)
         readonly.scoping(&block)
       end
     ensure
-      @disallow_replica -= 1 if which == :master
+      @disallow_replica -= 1 if which == :primary
```

Comment:
`on_cx_switch_block` and `on_primary_or_replica` are public methods, this is a breaking change. How about this instead?

```suggestion
      @disallow_replica -= 1 if [:primary, :master].include?(which)
```